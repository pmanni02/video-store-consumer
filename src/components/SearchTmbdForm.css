form button {
  padding-left: 2em;
  padding-right: 2em;
  padding-top: .25em;
  padding-bottom: .25em;
  margin-left: 1em;
  margin-right: 1em;
}

form button:hover {
  background-color: #56667A;
  color: white;
  cursor: pointer;
}
