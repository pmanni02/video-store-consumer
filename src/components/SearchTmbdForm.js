import React, { Component } from 'react';
import PropTypes from 'prop-types';
import '../App.css';
import './SearchTmbdForm.css';

class SearchTmbdForm extends Component {
  constructor(){
    super();
    this.state = {
      search: '',
    }
  }

  onFieldChange = (event) => {
    const search = event.target.value;
    this.setState({
      search,
    });
  }

  onFormSubmit = (event) => {
    event.preventDefault();

    // call separate function for axios call to Tmbd
    this.props.searchCallback(this.state.search);

    // clear form
    this.setState({
      search: '',
    });
  }


  render(){
    return(
      <form onSubmit = { this.onFormSubmit } >
        <div>
           <label
             htmlFor = "text"><strong>Search: </strong> 
           </label>
           <input name = "text" value = { this.state.search } type = "text"
             onChange = { this.onFieldChange }
           />
           <button type = "submit">Search TMBD</button>
         </div>

      </form>
    );
  }
}

SearchTmbdForm.propTypes = {
  searchCallback: PropTypes.func.isRequired
};

export default SearchTmbdForm;
