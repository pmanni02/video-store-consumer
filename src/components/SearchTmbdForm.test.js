import React from 'react';
import { shallow } from 'enzyme';
import SearchTmbdForm from './SearchTmbdForm';

describe('SearchTmbdForm', () => {
  test('snapshot', () => {
    const wrapper = shallow(
      <SearchTmbdForm
        searchCallback={()=>{}}
      />
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });
});
