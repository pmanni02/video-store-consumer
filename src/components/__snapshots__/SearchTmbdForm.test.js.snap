// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchTmbdForm snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div>
    <label
      htmlFor="text"
    >
      <strong>
        Search: 
      </strong>
    </label>
    <input
      name="text"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Search TMBD
    </button>
  </div>
</form>
`;
